cpp/src/Ice/InputStream.cpp

```diff
@@ -769,48 +763,6 @@ Ice::InputStream::read(vector<int16_t>& v)
     }
 }
 
-void
-Ice::InputStream::read(pair<const short*, const short*>& v)
-{
-    int32_t sz = readAndCheckSeqSize(static_cast<int>(sizeof(int16_t)));
-    if (sz > 0)
-    {
-#ifdef ICE_UNALIGNED
-        v.first = reinterpret_cast<int16_t*>(i);
-        i += sz * static_cast<int>(sizeof(int16_t));
-        v.second = reinterpret_cast<int16_t*>(i);
-#else
-
-        auto result = new short[static_cast<size_t>(sz)];
-        _deleters.push_back([result] { delete[] result; });
-        v.first = result;
-        v.second = result + sz;
-
-        Container::iterator begin = i;
-        i += sz * static_cast<int>(sizeof(int16_t));
-        if constexpr (endian::native == endian::big)
-        {
-            const byte* src = &(*begin);
-            byte* dest = reinterpret_cast<byte*>(&result[0]) + sizeof(int16_t) - 1;
-            for (int j = 0; j < sz; ++j)
-            {
-                *dest-- = *src++;
-                *dest-- = *src++;
-                dest += 2 * sizeof(int16_t);
-            }
-        }
-        else
-        {
-            copy(begin, i, reinterpret_cast<byte*>(&result[0]));
```

Comment:
Aren't we losing this optimization for ARM64?

Reply:
No, we still use this optimization. `template<typename T> void read(std::pair<const T*, const T*>& v)` for int16_t (there is some inconsistency in this file!) calls `Ice::InputStream::read(vector<int16_t>& v)` which has the same optimization:
```cpp
        if constexpr (endian::native == endian::big)
        {
            ...
        }
        else
        {
            copy(begin, i, reinterpret_cast<byte*>(&v[0]));
        }
```